---
title: "BINF 6309 Module 11 Assignment"
output:
  html_document:
    df_print: paged
---

### Setup 

```{r}

library(maftools)
library(R.utils)

```


# Reading and summarising MAF files

```{r}

#path to TCGA LAML MAF file
laml.maf = system.file('extdata', 'tcga_laml.maf.gz', package = 'maftools') 
#clinical information containing survival information and histology. This is optional
laml.clin = system.file('extdata', 'tcga_laml_annot.tsv', package = 'maftools') 

laml = read.maf(maf = laml.maf, clinicalData = laml.clin)

#Typing laml shows basic summary of MAF file.
laml

```


```{r}

#Shows sample summry.
getSampleSummary(laml)
#Shows gene summary.
getGeneSummary(laml)
#shows clinical data associated with samples
getClinicalData(laml)
#Shows all fields in MAF
#getFields(laml)
#Writes maf summary to an output file with basename laml.
write.mafSummary(maf = laml, basename = 'laml')

```


# Visualizing

### plotting MAF summary

number of variants and classifications in barcharts and boxplot.

```{r}

plotmafSummary(maf = laml, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)

```


### Oncoplots

Also known as waterfall plots. Comprehensive view of MAF file in single plot.

```{r}

#oncoplot for top ten mutated genes.
oncoplot(maf = laml, top = 10)

```


### Transitions and Transversions

Shows the specific percentage of base changes.

```{r}

laml.titv = titv(maf = laml, plot = FALSE, useSyn = TRUE)
#plot titv summary
plotTiTv(res = laml.titv)

```


### Lollipop plots for amino acid changes

Requires Amino acid change information, which is not required and thus does not have fixed column name. Default looks for AAChange, here we use Protein_Change. Default plot uses the longest isoform of the gene.

Lollipop plot is ____

```{r}

#lollipop plot for DNMT3A, which is one of the most frequent mutated gene in Leukemia.
lollipopPlot(
  maf = laml,
  gene = 'DNMT3A',
  AACol = 'Protein_Change',
  showMutationRate = TRUE,
  labelPos = 882
)

```

Can also plot custom data instead of MAF file - must be 2 column DF with pos and counts. General protein domains can be plotted with plotProtein(gene = , refSeqID = )


### Rainfall plots

Tumors are characterized by genomic loci with localized hyper-mutations (abnormally frequent mutation), which can be viewed by plotting inter variant distance on a linear genomic scale. Kataegis are defined as genomic segments containing six or more consecutive mutations with an average inter-mutation distance of less than 1k bp.

```{r}

brca <- system.file("extdata", "brca.maf.gz", package = "maftools")
brca = read.maf(maf = brca, verbose = FALSE)
rainfallPlot(maf = brca, detectChangePoints = TRUE, pointSize = 0.4)

```


### Compare mutation load against TCGA cohorts

For visualizing mutation load and burden vs the MC3 public MAF (33 other cohorts).
Mutation load- total result from accumulated problematic mutations.
Mutation burden- total number of mutations.

```{r}

laml.mutload = tcgaCompare(maf = laml, cohortName = 'Example-LAML', logscale = TRUE, capture_size = 50)

```


### Plotting VAF

Variant allele frequency as boxplot for estimating clonal status of top mutated genes. Clonal genes would have allele frequency around 50% assuming pure sample.

```{r}

plotVaf(maf = laml, vafCol = 'i_TumorVAF_WU')


```


# Analysis

### Somatic interactions

pair-wise Fisherâ€™s Exact test to detect significant pairs of genes

```{r}

#exclusive/co-occurance event analysis on top 10 mutated genes. 
somaticInteractions(maf = laml, top = 25, pvalue = c(0.05, 0.1))

```


### Cancer driver genes based on positional clustering

size of points proportional to number of clusters found in the gene.

```{r}

laml.sig = oncodrive(maf = laml, AACol = 'Protein_Change', minMut = 5, pvalMethod = 'zscore')
head(laml.sig)
plotOncodrive(res = laml.sig, fdrCutOff = 0.1, useFraction = TRUE, labelSize = 0.5)

```


### adding and summarising pfam domains

amino acid information.

```{r}

laml.pfam = pfamDomains(maf = laml, AACol = 'Protein_Change', top = 10)

#Protein summary (Printing first 7 columns for display convenience)
laml.pfam$proteinSummary[,1:7, with = FALSE]

#Domain summary (Printing first 3 columns for display convenience)
laml.pfam$domainSummary[,1:3, with = FALSE]


```


### survival analysis

will estimate probability of survival through the future.

```{r}

#Survival analysis based on grouping of DNMT3A mutation status
mafSurvival(maf = laml, genes = 'DNMT3A', time = 'days_to_last_followup', Status = 'Overall_Survival_Status', isTCGA = TRUE)

#Using top 20 mutated genes to identify a set of genes (of size 2) to predict poor prognostic groups
prog_geneset = survGroup(maf = laml, top = 20, geneSetSize = 2, time = "days_to_last_followup", Status = "Overall_Survival_Status", verbose = FALSE)

print(prog_geneset)

mafSurvGroup(maf = laml, geneSet = c("DNMT3A", "FLT3"), time = "days_to_last_followup", Status = "Overall_Survival_Status")

```


### comparing cohorts (two MAFs)

to understand differentially mutated genes between difference cancer cases or subtypes.

```{r}

#Primary APL MAF
primary.apl = system.file("extdata", "APL_primary.maf.gz", package = "maftools")
primary.apl = read.maf(maf = primary.apl)
#Relapse APL MAF
relapse.apl = system.file("extdata", "APL_relapse.maf.gz", package = "maftools")
relapse.apl = read.maf(maf = relapse.apl)
#Considering only genes which are mutated in at-least in 5 samples in one of the cohort to avoid bias due to genes mutated in single sample.
pt.vs.rt <- mafCompare(m1 = primary.apl, m2 = relapse.apl, m1Name = 'Primary', m2Name = 'Relapse', minMut = 5)
print(pt.vs.rt)

forestPlot(mafCompareRes = pt.vs.rt, pVal = 0.1)

genes = c("PML", "RARA", "RUNX1", "ARID1B", "FLT3")
coOncoplot(m1 = primary.apl, m2 = relapse.apl, m1Name = 'PrimaryAPL', m2Name = 'RelapseAPL', genes = genes, removeNonMutated = TRUE)

coBarplot(m1 = primary.apl, m2 = relapse.apl, m1Name = "Primary", m2Name = "Relapse")

lollipopPlot2(m1 = primary.apl, m2 = relapse.apl, gene = "PML", AACol1 = "amino_acid_change", AACol2 = "amino_acid_change", m1_name = "Primary", m2_name = "Relapse")

```


### Clinical enrichment analysis

enrichment based on clinical features - that is what mutations are most correlated to clinical groupings.

```{r}

fab.ce = clinicalEnrichment(maf = laml, clinicalFeature = 'FAB_classification')

#Results are returned as a list. Significant associations p-value < 0.05
fab.ce$groupwise_comparision[p_value < 0.05]

plotEnrichmentResults(enrich_res = fab.ce, pVal = 0.05, geneFontSize = 0.5, annoFontSize = 0.6)


```


### Drug-Gene interactions

checks for drug-gene interactions for druggability check purposes.

```{r}

dgi = drugInteractions(maf = laml, fontSize = 0.75)

dnmt3a.dgi = drugInteractions(genes = "DNMT3A", drugs = TRUE)

```


### Oncogenic signaling pathways

checks pathway enrichment based on TCGA cohorts.

```{r}
# could not run this section 
#pws = pathways(maf = laml, plotType = 'treemap')
#plotPathways(maf = laml, pathlist = pws)

```


# References

Mayakonda A, Lin DC, Assenov Y, Plass C, Koeffler HP. 2018. Maftools: efficient and comprehensive analysis of somatic variants in cancer. Genome Resarch. PMID: 30341162





